go/store/nbs/journal_writer.go

```diff
@@ -133,11 +146,50 @@ func createJournalWriter(ctx context.Context, path string) (wr *journalWriter, e
 		return nil, fmt.Errorf("expected file journalOffset 0, got %d", o)
 	}
 
-	return &journalWriter{
+	sendEg, _ := errgroup.WithContext(ctx)
+	recvEg, ctx := errgroup.WithContext(ctx)
+	indexChan := make(chan any, indexChanSize)
+	wr = &journalWriter{
 		buf:     make([]byte, 0, journalWriterBuffSize),
 		journal: f,
 		path:    path,
-	}, nil
+		recvEg:  recvEg,
+		sendEg:  sendEg,
+		indexCh: indexChan,
+	}
+	recvEg.Go(func() error {
+		return wr.recvIndexRecords(ctx, indexChan)
+	})
+
+	return wr, nil
+}
+
+func (wr *journalWriter) recvIndexRecords(ctx context.Context, c chan any) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return context.Cause(ctx)
+		case obj, ok := <-c:
+			if !ok {
+				return nil
+			}
+			var err error
+			switch l := obj.(type) {
+			case lookup:
+				wr.batchCrc = crc32.Update(wr.batchCrc, crcTable, l.a[:])
+				err = writeIndexLookup(wr.indexWriter, l)
+			case lookupMeta:
+				l.checkSum = wr.batchCrc
+				wr.batchCrc = 0
```

Comment:
Can this become a local variable of this goroutine func? If not, where else is accessing it and why its current access from that place threadsafe?